repaygithub/react-credit-card
# CACTUS-712 :: Dependency Upgrades

https://repayonline.atlassian.net/browse/CACTUS-712

Didn't have to make too many manual changes, even with major version upgrades.

I remediated all of the security vulnerabilities I could. There were a few that I couldn't do anything about, but they're all related to storybook or jest, so I'm not too worried about them.

### Testing
1. `cd` into the root of this repo and run `yarn`.
2. Run `yarn clean && yarn build`.
3. Run `yarn test`. This will run the linter, test types, and run unit tests.
4. Run `yarn dev` to start storybook.
5. Play around with the card in the stories. Edit each field, make sure the focus prop is working, check the boxes for the boolean props.

### Optional Extended Testing
To make sure the build isn't broken, you can follow these steps to publish the package and make a story for it in `cactus`.
1. Assuming you've already run `yarn build` from the testing instruction above, run `yalc publish` at the root of this repo.
2. Move over to `cactus` and `cd` into `modules/cactus-web`.
3. Run `yalc add @repay/react-credit-card && yarn`.
4. Open any component story file and copy/paste the following:
  ```
import '@repay/react-credit-card/dist/react-credit-card.css'
import CreditCard from '@repay/react-credit-card'
```
```
const focusTypes = ['number', 'cvc', 'expiration', 'name']
export const CreditCardTest = (): React.ReactElement => (
  <CreditCard
    number={text('number', '4111111111111111')}
    name={text('name', 'CS Human')}
    focused={select('focused', focusTypes, 'number')}
    expiration={text('expiration', '01/27')}
    cvc={text('cvc', '')}
    hasRadialGradient={boolean('useRadialGradient', false)}
    hasShadow={boolean('hasShadow', false)}
  />
)
```
5. Run `yarn start` and open the new story you just created. Make sure the card works there too.
